src/frontend/screens/Settings/MapSettings/MapImportBottomSheet.tsx

```diff
@@ -0,0 +1,322 @@
+ */
+const SelectImportContent = ({
+  onPressImport,
+  onPressClose,
+}: {
+  onPressImport: () => void;
+  onPressClose: () => void;
+}) => {
+  const { formatMessage: t } = useIntl();
+  return (
+    <BottomSheetContent
+      title={t(m.BackgroundMapTitle)}
+      buttonConfigs={[
+        {
+          onPress: onPressClose,
+          text: t(m.close),
+          variation: "outlined",
+        },
+      ]}
+    >
+      <TouchableOpacity onPress={onPressImport} style={styles.importButton}>
+        <View style={styles.importTextAndIcon}>
+          <MaterialIcon name="file-upload" size={30} color={MEDIUM_GREY} />
+          <Text style={styles.text}> {t(m.importFromFile)}</Text>
+        </View>
+        <Text style={[styles.text, { textAlign: "center" }]}>
+          {"( .mbtiles )"}
+        </Text>
+      </TouchableOpacity>
+    </BottomSheetContent>
+  );
+};
+
+/**
+ * Bottom sheet content for when an imported file is processing, checking the
+ * format etc. is correct (shows until the import endpoint returns with the
+ * import id). User is unable to close this sheet.
+ */
+const ImportingContent = () => {
+  return <BottomSheetContent title="Processing file..." buttonConfigs={[]} />;
+};
+
+/**
+ * Bottom sheet content for when there is an error with the selected file
+ */
+const FileErrorContent = ({ onPressClose }: { onPressClose: () => void }) => {
+  const { formatMessage: t } = useIntl();
+  return (
+    <BottomSheetContent
+      title={t(m.importErrorTitle)}
+      icon={<ErrorIcon color={RED} size={90} style={styles.errorIcon} />}
+      description={t(m.fileErrorDescription)}
+      buttonConfigs={[
+        {
+          onPress: onPressClose,
+          text: t(m.close),
+          variation: "outlined",
+        },
+      ]}
+    />
+  );
+};
+
+type BottomSheetStates = "select" | "importing" | "error";
+
+export interface MapImportBottomSheetMethods {
+  /** Open the bottom sheet by expanding it to its maximum snap point */
+  open: () => void;
+}
+
+/** Custom backdrop that will close sheet when pressed */
+const BackdropPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"close"} />
+  );
+};
+
+/** Custom backdrop that will not close sheet when pressed */
+const BackdropNonPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"none"} />
+  );
+};
+
+const MapImportBottomSheet = React.forwardRef<MapImportBottomSheetMethods, {}>(
+  (_props, ref) => {
+    const [state, setState] = React.useState<BottomSheetStates>("select");
+    const bottomSheetRef = React.useRef<BottomSheetMethods>(null);
+    const { snapPoints, updateSheetHeight } = useSnapPointsCalculator();
+    const { add: addMapImport } = useMapImportsManager();
+
+    React.useImperativeHandle(ref, () => {
+      return {
+        open: () => bottomSheetRef.current?.expand(),
+      };
+    });
```

Comment:
Is there a reason you didn't opt to just use the `bottomSheetProvider` and associated hook?

Reply:
that provider is only applicable for the bottom sheet modal component, but this is using the normal bottom sheet component. reasoning for that is due to a really subtle bug when trying to use our custom bottom sheet modal implementation, and using the normal bottom sheet worked. not an ideal divergence but maybe something to revisit later rather than perfecting here